packaging_automation/common_tool_methods.py

```diff
@@ -290,3 +323,154 @@ def remove_cloned_code(exec_path: str):
         except:
             print(f"Some files could not be deleted in directory {exec_path}. "
                   f"Please delete them manually or they will be deleted before next execution")
+
+
+def process_template_file(project_version: str, templates_path: str, template_file_path: str):
+    ''' This function gets the template files, changes tha parameters inside the file and returns the output.
+        Template files are stored under packaging_automation/templates and these files include parametric items in the
+        format of {{parameter_name}}. This function is used while creating docker files and pgxn files which include
+        "project_name" as parameter. Example usage is in "test_common_tool_methods/test_process_template_file".
+        Jinja2 is used as th the template engine and render function gets the file change parameters in the file
+         with the given input parameters and returns the output.'''
+    minor_version = get_minor_project_version(project_version)
+    env = get_template_environment(templates_path)
+    template = env.get_template(template_file_path)
+    return f"{template.render(project_version=project_version, project_minor_version=minor_version)}\n"
+
+
+def write_to_file(content: str, dest_file_name: str):
+    with open(dest_file_name, "w") as writer:
+        writer.write(content)
+
+
+def get_gpg_fingerprints_by_name(name: str) -> List[str]:
+    '''Returns GPG fingerprint by its unique key name. We use this function to determine the fingerprint that
+       we should use when signing packages'''
+    result = subprocess.run(shlex.split(f"gpg --list-keys "), check=True, stdout=subprocess.PIPE)
+    lines = result.stdout.decode("ascii").splitlines()
+    finger_prints = []
+    previous_line = ""
+    for line in lines:
+        if line.startswith("uid") and name in line:
+            finger_prints.append(previous_line.strip())
+            continue
+        previous_line = line
+    return finger_prints
+
+
+def delete_gpg_key_by_name(name: str, key_type: GpgKeyType):
+    keys = get_gpg_fingerprints_by_name(name)
+
+    # There could be more than one key with the same name. While statement is used to delete all the public keys
+    # until no key remains (i.e. key_id is empty).
+    # Public and private keys are stored with the same fingerprint. In some cases one of them may not be exist.
+    # Therefore non-existence case is possible
```

Comment:
outdated comment
```suggestion
```

Reply:
Instead of while I used for in the comment. Other than this issue, I think this comment is still applicable